neo/test/actor/RequestRouterTest.scala

```diff
@@ -1,30 +1,221 @@
-      // TODO need to be implemented
+    "parse transform and pass transform to berry" in {
+
+      val berryResult = Json.obj("result" -> "result example")
+      val frontEnd = new TestProbe(system)
+      val clientProps = Props(new Actor {
+        override def receive: Receive = {
+          case berryRequest: Request =>
+            frontEnd.ref ! berryRequest.postTransform.transform(berryResult)
+          case _ =>
+            throw new IllegalArgumentException("Message type sent to berry is not correct.")
+        }
+      })
+      val router = system.actorOf(RequestRouter.props(frontEnd.ref, clientProps, Config.Default))
+
+      frontEnd.send(router, nonSlicingRequestWithNoTransform)
+      frontEnd.expectMsg(berryResult)
+
+      frontEnd.send(router, slicingRequestWithNoTransform)
+      frontEnd.expectMsg(berryResult)
+
+      frontEnd.send(router, nonSlicingRequestWithWrapTransform)
+      frontEnd.expectMsg(Json.obj(
+        "key" -> "sample",
+        "value" -> berryResult
+      ))
+
+      frontEnd.send(router, slicingRequestWithWrapTransform)
+      frontEnd.expectMsg(Json.obj(
+        "key" -> "batch",
+        "value" -> berryResult
+      ))
+
+      ok
+    }
+
+    "distribute non-slicing and slicing requests to different berry clients" in {
+
+      val nonSlicingBerryResult = Json.obj("result" -> "non slicing result")
+      val slicingBerryResultOne = Json.obj("result" -> "slicing result 1")
+      val slicingBerryResultTwo = Json.obj("result" -> "slicing result 2")
+      val slicingBerryResultThree = Json.obj("result" -> "slicing result 3")
+
+      val frontEnd = new TestProbe(system)
+      val clientProps = Props(new Actor {
+        override def receive: Receive = {
+          case berryRequest: Request =>
+            (berryRequest.json \\ "sliceMillis").isEmpty match {
+              case true =>
+                frontEnd.ref ! nonSlicingBerryResult
+              case false =>
+                frontEnd.ref ! slicingBerryResultOne
+                Thread.sleep(500)
+                frontEnd.ref ! slicingBerryResultTwo
+                Thread.sleep(500)
+                frontEnd.ref ! slicingBerryResultThree
+            }
+          case _ =>
+            throw new IllegalArgumentException("Message type sent to berry is not correct.")
+        }
+      })
+      val router = system.actorOf(RequestRouter.props(frontEnd.ref, clientProps, Config.Default))
+
+      frontEnd.send(router, nonSlicingRequestWithNoTransform)
+      frontEnd.expectMsg(nonSlicingBerryResult)
+
+      frontEnd.send(router, slicingRequestWithNoTransform)
+      frontEnd.expectMsg(slicingBerryResultOne)
+      frontEnd.expectMsg(slicingBerryResultTwo)
+      frontEnd.expectMsg(slicingBerryResultThree)
+
+      frontEnd.send(router, slicingRequestWithNoTransform)
+      frontEnd.expectMsg(100 millis, slicingBerryResultOne)
+      frontEnd.send(router, slicingRequestWithNoTransform)
+
+      /*
+      ideal test case result:
```

Comment:
because the `clientProps` that you defined is not as the same as the `berryClient`. The `berryClient` itself will cancel the previous query.  It's been tested. So for here, we need to produce a real `berryClient`, and only mock the `ws` connection. However, it will be an integration test not a unit test. For this `router`, as long as the msg is delivered to berry it should be good.